pkg/daemon/kernelargs.go

```diff
@@ -93,63 +90,62 @@ func parseTuningFile(tuningFilePath, cmdLinePath string) ([]types.TuneArgument,
-				return addArguments, deleteArguments, err
+
+		var key string
+		var operation string
+		for _, k := range []string{"ADD", "DELETE"} {
+			if strings.HasPrefix(line, k) {
+				key = strings.TrimPrefix(fmt.Sprintf("%s ", k), k)
+				operation = strings.TrimSpace(k)
+				glog.V(2).Infof("Requested to %s kernel argument %s", operation, key)
+				break
 			}
-			if tuneableKarg {
-				// Find out if the argument is in use
-				inUse, err := isArgInUse(key, cmdLinePath)
-				if err != nil {
-					return addArguments, deleteArguments, err
-				}
-				if !inUse {
-					addArguments = append(addArguments, types.TuneArgument{Key: key, Bare: true})
-				} else {
-					glog.Infof(`skipping "%s" as it is already in use`, key)
-				}
+		}
+
+		if key == "" {
+			glog.Warningf("Malformed or unknown kernel arg directive %q, skipping", line)
+			continue
+		}
+
+		// NOTE: Today only specific bare kernel arguments are allowed so
+		// there is not a need to split on =.
+		tuneableKarg, err := isArgTunable(key)
+		if err != nil {
+			return addArguments, deleteArguments, err
+		}
+		if !tuneableKarg {
+			glog.Infof("Skipping unsupported kernel tunable %q", key)
+			continue
+		}
+
+		// Find out if the argument is in use
+		inUse, err := isKernelArgInUse(key)
+		if err != nil {
+			return addArguments, deleteArguments, err
+		}
+
+		if operation == "ADD" {
+			if !inUse {
+				addArguments = append(addArguments, types.TuneArgument{Key: key, Bare: true})
 			} else {
-				glog.Infof("%s not an allowlisted kernel argument", key)
+				glog.Infof("Kernal Argument %q is already defined, skipping addition", key)
 			}
-		} else if strings.HasPrefix(line, "DELETE ") {
-			// NOTE: Today only specific bare kernel arguments are allowed so
-			// there is not a need to split on =.
-			key := strings.TrimSpace(line[len("DELETE "):])
-			tuneableKarg, err := isArgTunable(key)
-			if err != nil {
-				return addArguments, deleteArguments, err
-			}
-			if tuneableKarg {
-				inUse, err := isArgInUse(key, cmdLinePath)
-				if err != nil {
-					return addArguments, deleteArguments, err
-				}
-				if inUse {
-					deleteArguments = append(deleteArguments, types.TuneArgument{Key: key, Bare: true})
-				} else {
-					glog.Infof(`skipping "%s" as it is not present in the current argument list`, key)
-				}
+		}
+		if operation == "DELETE" {
+			if inUse {
+				deleteArguments = append(deleteArguments, types.TuneArgument{Key: key, Bare: true})
 			} else {
-				glog.Infof("%s not an allowlisted kernel argument", key)
+				glog.Infof("Kernel argument %q is not currently used, skipping removal", key)
 			}
-		} else {
-			glog.V(2).Infof(`skipping malformed line in %s: "%s"`, tuningFilePath, line)
 		}
 	}
 	return addArguments, deleteArguments, nil
 }
 
 // UpdateTuningArgs executes additions and removals of kernel tuning arguments
-func UpdateTuningArgs(tuningFilePath, cmdLinePath string) (bool, error) {
-	if cmdLinePath == "" {
-		cmdLinePath = CmdLineFile
```

Comment:
Note: the use of `/proc/cmdline` is incorrect as it shows the kernel arguments for the current boot, not the kernel arguments for the next boot. `rpm-ostree` updates the BLSConfig and reads from it.